xline-client/tests/kv.rs

```diff
@@ -0,0 +1,216 @@
+    }
+
+    // delete a range of keys
+    {
+        let resp = client
+            .delete(DeleteRangeRequest::new("del11").with_range_end("del22"))
+            .await?;
+        assert_eq!(resp.deleted, 2);
+        let resp = client
+            .range(
+                RangeRequest::new("del11")
+                    .with_range_end("del22")
+                    .with_count_only(true),
+            )
+            .await?;
+        assert_eq!(resp.count, 0);
+    }
+
+    // delete key with prefix
+    {
+        let resp = client
+            .delete(DeleteRangeRequest::new("del3").with_prefix())
+            .await?;
+        assert_eq!(resp.deleted, 2);
+        let resp = client
+            .range(RangeRequest::new("del32").with_count_only(true))
+            .await?;
+        assert_eq!(resp.count, 0);
+    }
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_txn() -> Result<()> {
+    let mut cluster = Cluster::new(3).await;
+    cluster.start().await;
+    let mut client = Client::connect(cluster.addrs().clone(), ClientOptions::default())
+        .await?
+        .kv_client();
+
+    client.put(PutRequest::new("txn01", "01")).await?;
+
+    // transaction 1
+    {
+        let resp = client
+            .txn(
+                Txn::new()
+                    .when(&[Compare::value("txn01", CompareResult::Equal, "01")][..])
+                    .and_then(
+                        &[TxnOp::put(
+                            PutRequest::new("txn01", "02").with_prev_kv(true),
+                        )][..],
+                    )
+                    .or_else(&[TxnOp::range(RangeRequest::new("txn01"))][..]),
+            )
+            .await?;
+
+        assert!(resp.succeeded);
+        let op_responses = resp.responses;
+        assert_eq!(op_responses.len(), 1);
+
+        match op_responses[0].response.as_ref().unwrap() {
+            xlineapi::Response::ResponsePut(resp) => {
+                assert_eq!(resp.prev_kv.as_ref().unwrap().value, b"01")
+            }
+            _ => panic!("unexpected response"),
+        }
+
+        let resp = client.range(RangeRequest::new("txn01")).await?;
+        assert_eq!(resp.kvs[0].key, b"txn01");
+        assert_eq!(resp.kvs[0].value, b"02");
+    }
+
+    // transaction 2
+    {
+        let resp = client
+            .txn(
+                Txn::new()
+                    .when(&[Compare::value("txn01", CompareResult::Equal, "01")][..])
+                    .and_then(&[TxnOp::put(PutRequest::new("txn01", "02"))][..])
+                    .or_else(&[TxnOp::range(RangeRequest::new("txn01"))][..]),
+            )
+            .await?;
+
+        assert!(!resp.succeeded);
+        let op_responses = resp.responses;
+        assert_eq!(op_responses.len(), 1);
+
+        match op_responses[0].response.as_ref().unwrap() {
+            xlineapi::Response::ResponseRange(resp) => {
+                assert_eq!(resp.kvs[0].value, b"02")
+            }
+            _ => panic!("unexpected response"),
```

Comment:
please tell what's expected here, and all the "panic unexpected response". Yes it tells the truth, but it's helpless when debugging the test code.

Reply:
done